### PR TITLE
Use all rpc independently of CHAIN_ID

### LINKED ISSUE #355: Wallet connect should support multiple networks
Currently, wallet connect will only work for the network specified by CHAIN_ID (Rinkeby for Dev, Mainnet for Prod env-t).
But we should try to provide wallet connect on all supported networks: 
- xDAI and Mainnet for Prod env-t.
-  Rinkeby, xDAI and Mainnet for Dev( test) env-t.

### PR DESCRIPTION
**Pull request by hpmaxi**
Closes #355 

For this to work we need that NoahZinsmeister/web3-react#185 be merged or we could fork the subpackage of anxolin fork and use as a dependency (`web3-react/walletconnect-connector`)

### CODE REVIEW

**Comment by josojo** [comment]
>For this to work we need that NoahZinsmeister/web3-react#185 be merged

The team never merged our PRs. Anxo is also from gnosis.
I think we should depend on the fork

**Comment by hpmaxi** [comment]
@josojo @anxolin  As far as I know we can't add subpackage as a dependency that lives in a branch from another repository.  Is it possible to have it in another repository just the subpackage that changed? Or maybe there is another way to do some kind of flattening of sub-packages but I couldn't found any.

**Comment by anxolin** [comment]
Hi @hpmaxi , nice to meet you :) 

This worked for us https://github.com/gnosis/gp-swap-ui/pull/436 we haven't done too many tests, but apparently works. We will leave the wallet connect thing a few days, cause now we have to focus on something else, but let us know if it works for you. We will resume our work with this in a few days.

**Comment by hpmaxi** [comment]
@anxolin Nice to meet you too. Great! Let me try it and if it works I will make this PR ready for review.

**Comment by hpmaxi** [comment]
@josojo I'm having some issues with my mobile wallet, so I couldn't test this in depth but anyways I'm marking it as ready for review to have some feedback from you.

**Comment by josojo** [comment]
>@josojo I'm having some issues with my mobile wallet, so I couldn't test this in depth but anyways I'm marking it as ready for review to have some feedback from you.

thanks. I will test it later this evening.

**Comment by josojo** [comment]
I tested it and its not yet working for me.
I get the following error message
image

**Comment by josojo** [comment]
Okay, I did some more testing and digging.

Currently, we can not use the customers wallet connect wallet: `Minerva`, as we are not sending the chain_id in the session:
https://github.com/blocknative/onboard/issues/471

@maxi, could please look into sending this chainId as well and then test it with the Minerva wallet?

File: src/connectors/index.ts [hunk 1]
```diff
@@ -38,20 +38,10 @@ export const injected = new InjectedConnector({
...
+} = { 4: NETWORK_URL_RINKEBY, 100: NETWORK_URL_XDAI, 1: NETWORK_URL_MAINNET }
 
-// The WalletConnectConnector only supports a single RPC when configured,
-// so we are going to use the RPC associated with the Chain_Id environment variable configuration
-// Doing this, we will support every network, including xDai
 export const walletconnect = new WalletConnectConnector({
-  rpc: {
-    ...rpcForWalletConnect,
-  },
+  rpc: rpcForWalletConnect,
```
Comment (josojo) [inline src/connectors/index.ts:44]: I think handing over many id's is risky, as some wallets will then deny the connection, if not all ids are supported

**Review by josojo — APPROVED** [review]
thanks!

**Pull request merged**